src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentMarkupEncodingPass.cs

```diff
@@ -45,7 +45,7 @@ private class Rewriter : IntermediateNodeWalker
         // 1. New lines (\r, \n), tabs (\t), angle brackets (<, >) - so they get rendered as actual new lines, tabs, brackets instead of &#xA;
         // 2. Any character outside the ASCII range
 
-        private static readonly char[] EncodedCharacters = new[] { '\r', '\n', '\t', '<', '>' };
+        private static readonly char[] EncodedCharacters = new[] { '\r', '\n', '\t', '<', '>', '\'' };
```

Comment:
Why is `'` special here, compared to most other javascript characters? What about `"`, for example, or even `` ` `` itself?

Reply:
Yeah, it's not special. It's a quick fix of the bug though.

Thinking about this more, I think it would be good to have all special characters listed here (those that `builder.AddContent` would encode). If such list can be even obtained.

Or go any other route I outlined in the issue.